### PR TITLE
operator-olm-update-csv

### PR DESCRIPTION
**Pull request by entgigi**

### CODE REVIEW

File: controllers/reconciliation/manager.go [hunk 1]
```diff
@@ -62,16 +67,41 @@ func (r *ReconcileManager) MainReconcile(ctx context.Context, req ctrl.Request)
...
+	if utils.IsOlmInstallation() {
+		r.statusUpdater.SetReconcileProcessingComponent(ctx, req.NamespacedName, "Csv")
+		legacyReconcile := legacy.NewLegacyReconcileManager(r.Client, r.Log)
+		if err = legacyReconcile.ReconcileClusterServiceVersion(ctx, req, images); err != nil {
+			r.statusUpdater.SetReconcileFailed(ctx, req.NamespacedName, "CsvReconciliationFailed")
+			return err
+		}
+	} else {
+		r.statusUpdater.SetReconcileProcessingComponent(ctx, req.NamespacedName, "ImageInfo")
+		legacyReconcile := legacy.NewLegacyReconcileManager(r.Client, r.Log)
```
Comment (firegloves) [inline controllers/reconciliation/manager.go]: This is the same instruction as line 79? Could we move them outside of the if else?

Comment (entgigi) [reply controllers/reconciliation/manager.go]: yep, i will make it common

Comment (entgigi) [reply controllers/reconciliation/manager.go]: done

File: legacy/reconcile_imageinfo.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+			r.Log.Info("ImageInfo configMap not found")
+			return nil
+		}
+		r.Log.Error(err, "Error get ImageInfo configMap")
+		return err
+	}
+
+	for key, value := range configMap.Data {
+		switch {
+		case key == "app-builder-6-4":
```
Comment (zonia3000) [inline legacy/reconcile_imageinfo.go]: why not using `switch(key)` instead of repeating `key ==`?

File: legacy/reconcile_csv.go [hunk 3]
```diff
@@ -19,10 +20,16 @@ func NewLegacyReconcileManager(client client.Client, log logr.Logger) *LegacyRec
...
+	req ctrl.Request,
+	appImages service.EntandoAppImages) error {
+
+	r.Log.Info("Starting ClusterServiceVersion reconciliation flow")
+
 	csvList := &csv.ClusterServiceVersionList{}
 
 	if err := r.Client.List(ctx, csvList); err != nil {
+		r.Log.Error(err, "Error List ClusterServiceVersion")
 		return nil
```
Comment (firegloves) [inline legacy/reconcile_csv.go]: return the error

File: legacy/reconcile_imageinfo.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+	configMap := &corev1.ConfigMap{}
+	var imageInfo types.NamespacedName = types.NamespacedName{
+		Name:      common.ImageInfoConfigMap,
+		Namespace: req.NamespacedName.Namespace,
+	}
+	if err := r.Client.Get(ctx, imageInfo, configMap); err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("ImageInfo configMap not found")
+			return nil
```
Comment (firegloves) [inline legacy/reconcile_imageinfo.go]: return the error here too

**Review by firegloves — APPROVED** [review]
LGTM

File: legacy/reconcile_imageinfo.go (hunk 2 above)
Comment (entgigi) [reply legacy/reconcile_imageinfo.go]: my bad i didn't know very well switch in golang

File: legacy/reconcile_csv.go (hunk 3 above)
Comment (entgigi) [reply legacy/reconcile_csv.go]: done

File: legacy/reconcile_imageinfo.go (hunk 4 above)
Comment (entgigi) [reply legacy/reconcile_imageinfo.go]: done

**Review by zonia3000 — APPROVED** [review]

File: legacy/reconcile_imageinfo.go (hunk 2 above)
Comment (entgigi) [reply legacy/reconcile_imageinfo.go]: done

**Review by zonia3000 — APPROVED** [review]

**Pull request merged**